tests/e2e/test_runner.go

```diff
@@ -10,27 +23,68 @@ type TestRunner struct {
 	testDriver TestCaseDriver
 	target     ExecutionTarget
 	verbose    bool
+	result     TestResult
+}
+
+type TestResult struct {
+	StartTime time.Time
+	Status    string
+	Duration  time.Duration
+	Result    string
+}
+
+func (result *TestResult) Started() {
+	result.StartTime = time.Now()
+	result.Status = TEST_STATUS_STARTED
+}
+
+func (res *TestResult) Failed() {
```

Comment:
Maybe we want to check whether the result was already set and reject if it's set twice? not sure if it matters, probably overkill though